### PR TITLE
Fix protocol handler retrieval thread CPU usage when no blocks are pr…

### PR DESCRIPTION
**Pull request by sydhds**
…ocessed

* [ ] document all added functions
* [ ] try in sandbox /simulation/labnet
  * [ ] if part of node-launch, checked using the `resync_check` flag
* [ ] unit tests on the added/changed features
  * [ ] make tests compile
  * [ ] make tests pass 
* [ ] add logs allowing easy debugging in case the changes caused problems
* [ ] if the API has changed, update the API specification

### CODE REVIEW

**Review by bilboquet — APPROVED** [review]
Solves the problem on my side.
Tracing the tokio loop from which this function is called shows that branches from the select are taken each second.

File: massa-protocol-worker/src/handlers/block_handler/retrieval.rs [hunk 1]
```diff
@@ -933,6 +933,9 @@ impl RetrievalThread {
             .expect("could not compute next block retrieval timer tick");
 
         if self.asked_blocks.is_empty() && self.block_wishlist.is_empty() {
+            // Note: in mainnet and before ICO, no blocks are processed but the timer need to be updated
```
Comment (damip) [inline massa-protocol-worker/src/handlers/block_handler/retrieval.rs:936]: not ICO but genesis, and it is not specific to mainnet

Comment (damip) [reply massa-protocol-worker/src/handlers/block_handler/retrieval.rs:936]: need => needs

**Review by damip — CHANGES_REQUESTED** [review]
Nice catch !
TODO: fix comment, shorten all thread names to fit

**Comment by AurelienFT** [comment]
> @sydhds @bilboquet @AurelienFT Does this require an urgent 1.1 deployment ? or does it have no impact at genesis ?

No it should stop when blocks will be produced. Could be cool to verify with a labnet

**Review by modship — APPROVED** [review]

**Comment by Floyz** [comment]
> > @sydhds @bilboquet @AurelienFT Does this require an urgent 1.1 deployment ? or does it have no impact at genesis ?
> 
> No it should stop when blocks will be produced. Could be cool to verify with a labnet

It has impact on node runners hardware tho, if you want people to have node running before genesis it would be great to publish a new release.
I stopped mine due to this bug, consuming one full CPU for nothing and impacting other services on a shared hardware. ( + CPU temp going high )

File: massa-protocol-worker/src/handlers/block_handler/retrieval.rs (hunk 1 above)
Comment (sydhds) [reply massa-protocol-worker/src/handlers/block_handler/retrieval.rs]: done

File: massa-node/src/survey.rs [hunk 2]
```diff
@@ -12,6 +12,10 @@ use tracing::info;
 // use std::time::Duration;
 use tracing::warn;
 
+// massa-survey
+const THREAD_NAME: &str = "massa-survey";
```
Comment (damip) [inline massa-node/src/survey.rs:16]: clippy

Comment (sydhds) [reply massa-node/src/survey.rs]: fixed

File: massa-node/src/survey.rs [hunk 3]
```diff
@@ -49,13 +49,16 @@ impl MassaSurvey {
         if massa_metrics.is_enabled() {
             #[cfg(all(not(feature = "sandbox"), not(test)))]
             {
+                // massa-survey
+                const THREAD_NAME: &str = "massa-survey";
```
Comment (bilboquet) [inline massa-node/src/survey.rs:53]:
```suggestion
                const THREAD_NAME: &str = "massa-survey";
```
?

Comment (bilboquet) [reply massa-node/src/survey.rs:53]: can't apply because of #cfg()

**Review by damip — APPROVED** [review]
Good !

**Pull request merged**